lib-extra/src/main/java/com/diffplug/spotless/extra/config/MavenCoordinates.java

```diff
@@ -0,0 +1,223 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.diffplug.spotless.extra.config;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.Serializable;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Set;
+import java.util.TreeSet;
+import java.util.stream.Collectors;
+
+/** Configuration of Maven coordinates */
+class MavenCoordinates implements Serializable {
+
+	// Not used, only the serialization output is required to determine whether the object has changed
+	private static final long serialVersionUID = 1L;
+	private final Set<Coordinate> coordinates;
+
+	/** Creates empty coordinate set */
+	MavenCoordinates() {
+		this.coordinates = new TreeSet<Coordinate>(new DependencyComparator());
+	}
+
+	/** Get all dependency restrictions */
+	String[] get() {
+		return coordinates.stream().map(c -> c.toString()).toArray(size -> new String[size]);
+	}
+
+	/** Add missing coordinates and overwrite existing ones */
+	void update(MavenCoordinates other) {
+		for (Coordinate coordinate : other.coordinates) {
+			update(coordinate);
+		}
+	}
+
+	/** Add missing coordinates and overwrite existing ones */
+	void update(String... coordinates) {
+		for (String coordinate : coordinates) {
+			update(new Coordinate(coordinate));
+		}
+	}
+
+	/** Add missing coordinates and overwrite existing ones */
+	void update(Coordinate... coordinates) {
+		for (Coordinate coordinate : coordinates) {
+			this.coordinates.remove(coordinate); //Update existing
+			this.coordinates.add(coordinate);
+		}
+	}
+
+	/**
+	 * Add coordinates if missing.
+	 * Expecting reference to ASCII content.
+	 * Each line must correspond to the {@link Coordinate#FORMAT}.
+	 */
+	void add(URL coordinates) {
+		try {
+			BufferedReader reader = new BufferedReader(new InputStreamReader(coordinates.openStream(), "UTF-8"));
+			try {
+				List<Coordinate> additionalCoordinates = reader.lines().map(l -> new Coordinate(l)).collect(Collectors.toList());
+				this.coordinates.addAll(additionalCoordinates);
+			} finally {
+				reader.close();
+			}
+		} catch (IOException e) {
+			throw new UserArgumentException(coordinates, "Cannot access URL for optaining Maven coordinates.", e);
+		}
+	}
+
+	/** Add coordinates if missing */
+	void add(Coordinate... coordinates) {
+		for (Coordinate coordinate : coordinates) {
+			this.coordinates.add(coordinate);
+		}
+	}
+
```

Comment:
Gradle up-to-date checking is just based on serialization, no need to implement equals or hashCode.  I think dangerous to implement because they aren't actually used / respected by the up-to-date check.